mu4/scenes/palette/internal/mu4paletteadapter.cpp

```diff
@@ -56,76 +56,24 @@ void MU4PaletteAdapter::showMasterPalette(const QString& arg)
     NOT_IMPLEMENTED;
 }
 
-void MU4PaletteAdapter::selectInstrument(Ms::InstrumentChange* i)
+bool MU4PaletteAdapter::isSelected() const
 {
-    Q_UNUSED(i);
-    NOT_IMPLEMENTED;
-}
-
-Ms::Score* MU4PaletteAdapter::currentScore() const
-{
-    NOT_IMPLEMENTED;
-    return nullptr;
-}
-
-Ms::ScriptRecorder* MU4PaletteAdapter::getScriptRecorder() const
-{
-    NOT_IMPLEMENTED;
-    return nullptr;
-}
-
-bool MU4PaletteAdapter::editMode() const
-{
-    NOT_IMPLEMENTED;
-    return false;
-}
-
-Ms::ScoreState MU4PaletteAdapter::mscoreState() const
-{
-    NOT_IMPLEMENTED;
-    return Ms::ScoreState::STATE_DISABLED;
-}
-
-void MU4PaletteAdapter::changeState(Ms::ViewState s)
-{
-    Q_UNUSED(s);
-    NOT_IMPLEMENTED;
-}
-
-void MU4PaletteAdapter::cmdAddSlur(const Ms::Slur* slurTemplate)
-{
-    Q_UNUSED(slurTemplate);
-    NOT_IMPLEMENTED;
-}
-
-void MU4PaletteAdapter::applyDrop(Ms::Score* score, Ms::Element* target, Ms::Element* e,
-                                  Qt::KeyboardModifiers modifiers,
-                                  QPointF pt, bool pasteMode)
-{
-    Q_UNUSED(score);
-    Q_UNUSED(target);
-    Q_UNUSED(e);
-    Q_UNUSED(modifiers);
-    Q_UNUSED(pt);
-    Q_UNUSED(pasteMode);
-
-    NOT_IMPLEMENTED;
-}
+    auto notation = globalContext()->currentNotation();
+    if (!notation) {
+        return false;
+    }
 
-void MU4PaletteAdapter::moveCursor()
-{
-    NOT_IMPLEMENTED;
+    return !notation->interaction()->selection()->isNone();
 }
 
-void MU4PaletteAdapter::setFocus()
+bool MU4PaletteAdapter::applyPaletteElement(Ms::Element* element, Qt::KeyboardModifiers modifiers)
```

Comment:
Well, I'm finally confused, what is the purpose of MU4 adapter?

Reply:
I'd expect "adapter" for the old implementation, so why we need adapter for the new one(mu4)?

Reply:
Old code in different places called different classes from `mscore`., For example, `MuseScore`, `ScoreView`...
The idea is that the old code calls the methods of the intermediary - the adapter, whose implementations are different for the old and new code.
My goal was to transfer the implementation of the palettes as is, without changing its implementation. Because Palettes are not one, two classes, this is a rather large and complex subsystem with a lot of functionality.